src/App.jsx

```diff
@@ -14,45 +17,106 @@ export const goodsFromServer = [
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
+
+let isSorted = false;
+
+const getSortedGoods = (goods, sortField, isReverse) => {
+  const sortedGoods = [...goods];
+
+  if (sortField) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          isSorted = true;
+
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          isSorted = true;
+
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  } else if (isReverse) {
+    isSorted = true;
+  } else {
+    isSorted = false;
+  }
+
+  return isReverse ? sortedGoods.reverse() : sortedGoods;
+};
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const visibleGoods = getSortedGoods(goodsFromServer, sortField, isReverse);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+          className={cn(
+            'button is-info',
+            { 'is-light': sortField !== SORT_FIELD_ALPHABET },
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          className={cn(
+            'button is-success',
+            { 'is-light': sortField !== SORT_FIELD_LENGTH },
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => (isReverse
+            ? setIsReverse(false)
+            : setIsReverse(true))}
+          className={cn(
+            'button is-warning',
+            { 'is-light': !isReverse },
+          )}
+        >
+          Reverse
+        </button>
+
+        {isSorted && (
+          <button
+            type="button"
+            onClick={() => {
+              setIsReverse(false);
+              setSortField('');
+            }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here